global_frame: map
robot_base_frame: base_link
update_frequency: 5.0
publish_frequency: 2.0
static_map: true
rolling_window: false
resolution: 0.05
transform_tolerance: 0.5

plugins:
  - {name: static_layer, type: "costmap_2d::StaticLayer"}
  - {name: slope_layer, type: "slope_costmap_layer::SlopeCostmapLayer"}
  - {name: inflation_layer, type: "costmap_2d::InflationLayer"}

static_layer:
  map_topic: /map
  subscribe_to_updates: true

slope_layer:
  enabled: true
  use_pcd_map: true
  pcd_map_path: /root/shared_dir/LIO-SAM/shangpo2map0.01/GlobalMap.pcd
  max_slope_angle: 30.0
  slope_threshold: 25.0
  slope_weight: 0.7
  max_roughness: 0.1
  roughness_threshold: 0.05
  roughness_weight: 0.3
  resolution: 0.05
  visualize: true

inflation_layer:
  enabled: true
  cost_scaling_factor: 10.0
  inflation_radius: 0.5 